### PR TITLE
Walrus diagnostic script and recipe

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
Adding a diagnostic script and recipe for preprocessing ERA5 data for the WALRUS model.

### CODE REVIEW

**Comment by JaroCamphuijsen** [comment]
This pull request is made by @JaroCamphuijsen to enable discussion on proceedings in the branch and in absence of @SarahAldt

File: WALRUS/diag_WALRUS.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        'domains': ['global'],
+        'authors': ['berg_pe'],
+        'references': ['acknow_project'],
+    }
+    diagnostic_file = get_diagnostic_filename(basename, cfg, extension)
+    with ProvenanceLogger(cfg) as provenance_logger:
+        provenance_logger.log(diagnostic_file, record)
+
+
+def shapeselect(cfg, cube):
```
Comment (JaroCamphuijsen) [inline WALRUS/diag_WALRUS.py:34]: You don't need to copy the shapeselect diagnostic into this one. It's possible to just call it from the recipe.

File: WALRUS/recipe_WALRUS.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      script1:
+        model: WALRUS
+        script: /path/to/your/WALRUS/diag_WALRUS.py
+        dataname: 'PEQ_Hupsel' 
+        model_calib: false
+        calibdata: 'PEQ_Hupsel.dat' 
+        geojfile: 'walrus-hupsel.geojson'
+        weighting_method: 'mean_inside'
+        convert_units: true
+        write_dat: true
```
Comment (JaroCamphuijsen) [inline WALRUS/recipe_WALRUS.yml:37]:
```suggestion
        write_dat: true

  diagnostic2:
    description: Extracting data for user defined shapefile.
    variables:
      tas:
      pr:
    scripts:
      script1:
        script: shapeselect/diag_shapeselect.py
        shapefile: 'Thames.shp'
        weighting_method: 'mean_inside'
        write_xlsx: true
        write_netcdf: true

```
Or switch with diag_WALRUS if needed

**Comment by Peter9192** [comment]
Migrated to https://github.com/ESMValGroup/ESMValTool/pull/1404

**Pull request closed without merging**